ortools/math_opt/solvers/xpress/g_xpress.cc

```diff
@@ -0,0 +1,263 @@
+}
+
+absl::Status Xpress::SetObjective(bool maximize, double offset,
+                                  const absl::Span<const int> colind,
+                                  const absl::Span<const double> values) {
+  RETURN_IF_ERROR(ToStatus(XPRSchgobjsense(
+      xpress_model_, maximize ? XPRS_OBJ_MAXIMIZE : XPRS_OBJ_MINIMIZE)))
+      << "Failed to change objective sense in XPRESS";
+
+  static int indexes[1] = {-1};
+  double xprs_values[1] = {-offset};
+  RETURN_IF_ERROR(ToStatus(XPRSchgobj(xpress_model_, 1, indexes, xprs_values)))
+      << "Failed to set objective offset in XPRESS";
+
+  const int n_cols = static_cast<int>(colind.size());
+  auto c_colind = const_cast<int*>(colind.data());
+  auto c_values = const_cast<double*>(values.data());
+  return ToStatus(XPRSchgobj(xpress_model_, n_cols, c_colind, c_values));
+}
+
+absl::Status Xpress::ChgCoeffs(absl::Span<const int> rowind,
+                               absl::Span<const int> colind,
+                               absl::Span<const double> values) {
+  const int n_coefs = static_cast<int>(rowind.size());
+  auto c_rowind = const_cast<int*>(rowind.data());
+  auto c_colind = const_cast<int*>(colind.data());
+  auto c_values = const_cast<double*>(values.data());
+  return ToStatus(
+      XPRSchgmcoef(xpress_model_, n_coefs, c_rowind, c_colind, c_values));
+}
+
+absl::StatusOr<int> Xpress::LpOptimizeAndGetStatus(std::string flags) {
+  RETURN_IF_ERROR(ToStatus(XPRSlpoptimize(xpress_model_, flags.c_str())))
+      << "XPRESS LP solve failed";
+  int xpress_status;
+  RETURN_IF_ERROR(
+      ToStatus(XPRSgetintattrib(xpress_model_, XPRS_LPSTATUS, &xpress_status)))
+      << "Could not get XPRESS status";
+  return xpress_status;
+}
+absl::Status Xpress::PostSolve() {
+  return ToStatus(XPRSpostsolve(xpress_model_));
+}
+
+absl::StatusOr<int> Xpress::MipOptimizeAndGetStatus() {
+  RETURN_IF_ERROR(ToStatus(XPRSmipoptimize(xpress_model_, nullptr)))
+      << "XPRESS MIP solve failed";
+  int xpress_status;
+  RETURN_IF_ERROR(
+      ToStatus(XPRSgetintattrib(xpress_model_, XPRS_MIPSTATUS, &xpress_status)))
+      << "Could not get XPRESS status";
+  return xpress_status;
+}
+
+void Xpress::Terminate() { XPRSinterrupt(xpress_model_, XPRS_STOP_USER); };
+
+absl::StatusOr<int> Xpress::GetIntAttr(int attribute) const {
+  int result;
+  RETURN_IF_ERROR(ToStatus(XPRSgetintattrib(xpress_model_, attribute, &result)))
+      << "Error getting Xpress int attribute: " << attribute;
+  return result;
+}
+
+absl::Status Xpress::SetIntAttr(int attribute, int value) {
+  return ToStatus(XPRSsetintcontrol(xpress_model_, attribute, value));
+}
+
+absl::StatusOr<double> Xpress::GetDoubleAttr(int attribute) const {
+  double result;
+  RETURN_IF_ERROR(ToStatus(XPRSgetdblattrib(xpress_model_, attribute, &result)))
+      << "Error getting Xpress double attribute: " << attribute;
+  return result;
+}
+
+absl::StatusOr<std::vector<double>> Xpress::GetPrimalValues() const {
```

Comment:
In which context is this function supposed to be called? There are only very special situations (inside callbacks) in which you would want to use `XPRSgetlpvalues()` to get the x vector. You most likely want to use [XPRSgetsolution()](https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSgetsolution.html) here. This also has the advantage that it returns a status that indicates whether a solution is available at all.